### PR TITLE
Support spaces in git repository path

### PR DESCRIPTION
**Pull request by davidstosik**
Fixes https://github.com/basecamp/kamal/issues/1036

~~I haven't written any tests for this though I'd be interested to try.~~

### CODE REVIEW

File: lib/kamal/commands/builder/clone.rb [hunk 1]
```diff
@@ -1,29 +1,31 @@
 module Kamal::Commands::Builder::Clone
-  extend ActiveSupport::Concern
-
-  included do
-    delegate :clone_directory, :build_directory, to: :"config.builder"
-  end
```
Comment (davidstosik) [inline lib/kamal/commands/builder/clone.rb:6]: This delegation becomes superfluous when we need to escape the values returned by `clone_directory` and `build_directory`, so I just removed it. Inlined `config.builder.clone_directory.shellescape` in the single location it's needed, and factorized `config.builder.build_directory.shellescape` in the `escaped_build_directory` method.

File: test/commands/builder_test.rb [hunk 2]
```diff
@@ -149,15 +149,26 @@ class CommandsBuilderTest < ActiveSupport::TestCase
...
+
   private
     def new_builder_command(additional_config = {})
       Kamal::Commands::Builder.new(Kamal::Configuration.new(@config.deep_merge(additional_config), version: "123"))
     end
 
-    def build_directory
-      "#{Dir.tmpdir}/kamal-clones/app/kamal/"
-    end
-
```
Comment (davidstosik) [inline test/commands/builder_test.rb:160]: I wondered what this was, and realized it has become unnecessary when tests still passed without it.

**Comment by davidstosik** [comment]
cc @djmb

**Comment by djmb** [comment]
Thanks @davidstosik!

**Pull request merged**